### DISCUSSION TITLE
The file upload attribute does not trigger wire:dirty.

### DISCUSSION THREAD

**Discussion by 3yekn1** [livewire/livewire#6576] [7. Bugs (archived)]
### Livewire version

^3.0@beta

### Which PHP version are you using?

PHP 8.2

### Steps To Reproduce

1. Select image to upload
2. The wire:dirty element is expected to be displayed, but in fact wire:dirty.remove is still displayed.

```php
    #[Rule('image|max:1024', as: '個人頭貼')]
    public $avatar;
```

```html
<input type="file" wire:model='avatar' id="avatar" class="form-control">

<button
        type="submit"
        class="btn btnMore"
        wire:dirty.remove
        style='border: 0;'
        disabled
    >Submit</button>

    <button
        type="submit"
        class="btn btnMore"
        wire:dirty
    >Submit</button>
```

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by dweerd** [comment]
I also ran into this. And another related thing: I have a `<div wire:loading>Saving...</div>` on my form, but this also shows the loading state while a file is (temporarily) uploaded. This makes some sense as it is temporarily saving something, but for the user it may be confusing as it looks like he doesn't have to hit 'Save' anymore to submit the form and make sure the temporarily saved file is permanently saved.

Update: I fixed the latter by targeting the save action using `wire:target="save"`.

**Comment by hardevine** [comment]
have you found the solution to 'file not triggering dirty flag' issue ?

**Comment by dvxgit-jsoriano** [comment]
Is this issue still unresolved? Is there a workaround to trigger the wire:dirty in a file input?

**Reply by hardevine** [reply]
you can use updated hooks
